File: Backend/Backend/DTOs/ChangeLogDto.cs
using Backend.Models;

namespace Backend.DTOs
{
    public class ChangeLogDto
    {
        public string Id { get; set; }
        public DateTime ChangeDate { get; set; }
        public string TaskItemId { get; set; }
        public AppUserDto Changer { get; set; }

        // Task properties
        public string OldTitle { get; set; }
        public string OldDetails { get; set; }
        public string OldPriority { get; set; }
        public string OldDeadline { get; set; }
        public string OldAssigneeId { get; set; }
        public AppUserDto OldAssignee { get; set; }
        public WorkflowItem OldWorkflowItem { get; set; }

        public string NewTitle { get; set; }
        public string NewDetails { get; set; }
        public string NewPriority { get; set; }
        public string NewDeadline { get; set; }
        public string NewAssigneeId { get; set; }
        public AppUserDto NewAssignee { get; set; }
        public WorkflowItem NewWorkflowItem { get; set; }
    }
}

File: Frontend/Frontend/src/app/_dtos/changeLogDto.ts
import { WorkflowItem } from "../_models/workflowitemmodel"
import { AppUserDto } from "./appUserDto"

export class ChangeLog{
    id: string = ''
    changeDate: string = ''
    taskItemId: string = ''
    changer: AppUserDto = new AppUserDto

    oldTitle: string = ''
    oldDetails: string = ''
    oldPriority: string = ''
    oldAssigneeId: string = ''
    oldAssignee: AppUserDto = new AppUserDto
    oldDeadline: string = ''
    oldWorkflowItem: WorkflowItem = new WorkflowItem

    newTitle: string = ''
    newDetails: string = ''
    newPriority: string = ''
    newAssigneeId: string = ''
    newAssignee: AppUserDto = new AppUserDto
    newDeadline: string = ''
    newWorkflowItem: WorkflowItem = new WorkflowItem
}